### ISSUE TITLE
igxForOf - Keyboard navigation with Tab does not work properly

### ISSUE THREAD

**Issue by lalo-mx** [IgniteUI/igniteui-angular#9835]
## Description  
Keyboard navigation with Tab does not work properly in virtualized list of inputs.

[Sample of igxForOf](https://stackblitz.com/edit/github-xpngpl?file=src%2Fapp%2FigxFor%2FigxFor.component.html)
 * igniteui-angular version: 12.0.7
 * browser:  Chrome  91.0.4472.124

## Steps to reproduce  

1. Open sample
2. Navigate to the  last input with keyboard(TAB)
3. The scroll bar disappears

## Result  
The scroll bar disappears

## Expected result  
The scroll bar moves along with the content when changing elements with the TAB key.

## Attachments  
[github-xpngpl.zip](https://github.com/IgniteUI/igniteui-angular/files/6780023/github-xpngpl.zip)

**Comment by ChronosSF** [comment]
@lalo-mx , it was never intended for igx-for-of to handle keyboard navigation and scroll the virtualized content. Instead its consumers should handle the keyboard presses as they see fit and use the igx-for-of navigate API to scroll it accordingly.

**Comment by lalo-mx** [comment]
@ChronosSF Thanks for reviewing the report, maybe it could be added as a known limitation of the component. What approach would you suggest? Handle al keyboard events? O maybe is it best to handle the focus event?

**Comment by lalo-mx** [comment]
@ChronosSF Is there a way to know that a virtualized item is generated but out of viewport? O maybe only generate visible items.

**Comment by ChronosSF** [comment]
I'd probably try with handling <kbd>TAB</kbd> and <kbd>SHIFT</kbd> + <kbd>TAB</kbd> (for going up). You preventDefault on the original event except when <kbd>SHIFT</kbd> + <kbd>TAB</kbd> on the first element or <kbd>TAB</kbd> on the last.

IgxForOf always renders 1 element in excess so that it can smooth scroll. So you can start going down when the index of the element that the user tabs from is `igxForOf.state.chunkSize - 2` . Then you navigate to index that is `igxForOf.state.startIndex +1`  and focus the next element which should actually have the index of the element  the user tabbed from before the scroll. 

You'll probably encounter other issues but this is the gist of it. Alternatively, we could also add keyboard navigation to igxList which supports igxForOf or you could use another navigatable component supporting templates and igxForOf such as https://www.infragistics.com/products/ignite-ui-angular/angular/components/drop-down-virtual . While not particularly pretty, it could be possible to just keep the dropdown open with CSS and use that instead of naked igxForOf. 

c.c. @radomirchev
